### PR TITLE
Feature/additional device settings

### LINKED ISSUE #582: Point-To-Point vpn configuration help
Hey there,

Thanks for continuing to maintain this is easy to get going and use. 

Im just wondering if there is any way to edit the generic `wg0.conf` that the server would use? looking in the source it looks to be abstracted away is that correct?

i was hoping to try something like this https://www.digitalocean.com/community/tutorials/how-to-create-a-point-to-point-vpn-with-wireguard-on-ubuntu-16-04

hoping that we can have 1 vpn configuration for clients to connect to as the "base/core" that then talks to other servers in other subnets/locations.

Or does adding another client via the ui do this anyway? the only issue i see is i need to be able to set the endpoint also which the ui doesn't support?

```
[Peer]
PublicKey = public_key_of_first_server
AllowedIPs = 10.0.0.1/32
Endpoint = public_IP_of_first_server:5555
```

### PR DESCRIPTION
**Pull request by phyzical**
Looks at adding support for device specific configuration of the endpoint and keepalive values per peer.

closes #582

 i got up to the logic around "addPeer" and found out its a submodule

https://github.com/search?q=repo%3AfreifunkMUC%2Fwg-embed%20addpeer&type=code

@GoliathLabs  is there any pointers on how i would pin this module to a branch to perform a full end to end test of the logic? im fairly green with go

https://github.com/freifunkMUC/wg-embed/pull/22

is working and changes are being reflected in the wg0 config as expected

Screenshot 2024-02-07 at 11 15 42 am

image can be tested via phyzical/wg-access-server:latest

### CODE REVIEW

**Comment by phyzical** [comment]
@GoliathLabs thx was able to jimmy rig it to work via mounting and using require, added rough docs to help others in the future

The only issue i am facing atm is that for some reason the Persistent keep alive value is not being pulled out of the database after being saved? this is a visual only bug though.

I have confirmed its in the db 
image

any idea if there is something magic i have missed? 
if i dump the json object returned from the api call i see its just missing?
image

**Comment by fbuetler** [comment]
@phyzical you will also need to add it [here](https://github.com/phyzical/wg-access-server/blob/feature/additional-device-settings/internal/services/server_service.go#L60-L79).

Any reason to not echo back the device endpoint [here](https://github.com/freifunkMUC/wg-access-server/pull/583/files#diff-3db472c4eeb87a7945b08b2d5f7df74d9bb9200724f7ef10d632b5f30415f9b7R20-R36)?

**Comment by fbuetler** [comment]
On another note, can you please revert the various format changes you did? It pollutes the PR and makes it hard to read IMO.

For reference, we need to first merge https://github.com/freifunkMUC/wg-embed/pull/22, release `wg-embed`  and update it in the `go.mod`.

File: proto/devices.proto [hunk 1]
```diff
@@ -32,12 +32,15 @@ message Device {
   string owner_email = 12;
   string owner_provider = 13;
   string preshared_key = 14;
+  int32 persistent_keepalive_interval = 15;
```
Comment (fbuetler) [inline proto/devices.proto:35]: Any reason to not echo back the device endpoint here?

**Comment by phyzical** [comment]
> @phyzical you will also need to add it [here](https://github.com/phyzical/wg-access-server/blob/feature/additional-device-settings/internal/services/server_service.go#L60-L79).

But this is device specific not server specific 

> On another note, can you please revert the various format changes you did? It pollutes the PR and makes it hard to read IMO.

@fbuetler its all auto lint conforming to this projects prettierrc i dont think i will even be able to undo those changes and still formatting anything else, it looks like if you hide whitespace in the review settings it becomes minimal again.

File: proto/devices.proto (hunk 1 above)
Comment (phyzical) [reply proto/devices.proto:35]: the endpoint was already being exposed on the device

**Comment by phyzical** [comment]
hmm i need to do more research, while this pr does let me connect two wg access-servers to each other i still cant resolve traffic between the two like i had hoped...

**Pull request closed without merging**